File: project_euler_study/0001/p_e_0001.exs
# Project Euler No.0001 by Elixir

defmodule Solution do

  defp sum_multiples_3or5(num) do
    1..(num - 1)
    |> Enum.filter(fn(x) -> (rem(x, 3) == 0) or (rem(x, 5) == 0) end)
    |> Enum.sum
  end

  def run() do
    sum_multiples_3or5(10)   |> IO.puts()
    sum_multiples_3or5(1000) |> IO.puts()
  end

end

Solution.run()

File: project_euler_study/0001/p_e_0001.erl
%%% Project Euler No.0001 by Erlang

-module(p_e_0001).
-export([run/0]).

sum_multiples_3or5(N) ->
    Num = [X || X <- lists:seq(1, N - 1), (X rem 3 == 0) or (X rem 5 == 0)],
    lists:sum(Num).

run() ->
    io:format("~p~n", [sum_multiples_3or5(10)]),
    io:format("~p~n", [sum_multiples_3or5(1000)]).

